napari/layers/base/base.py

```diff
@@ -1026,6 +1025,16 @@ def _type_string(self) -> str:
     def as_layer_data_tuple(self):
         state = self._get_state()
         state.pop('data', None)
+        if hasattr(self, '_deprecated_properties_list'):
```

Comment:
After some more thought, I find that this approach makes things hard to understand (because the attr is only added/mutated in `Points._add_deprecated_properties`) and forces all layer types to understand it, so I'm going to push a new commit that uses `DeprecatedProperty`, but isolates some of the changes to `Points` and without so much repetition.

I think we will want a general purpose way to handling these across layers, but I don't think this is quite right yet. Open to alternative implementations that are easier to understand.

Happy to discuss more in a community meeting!